python/tests/graph/test_builder.py

```diff
@@ -0,0 +1,157 @@
+#
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import print_function
+
+from glob import glob
+import os
+
+import numpy as np
+import tensorflow as tf
+import keras.backend as K
+from keras.applications import InceptionV3
+from keras.applications import inception_v3 as iv3
+from keras.preprocessing.image import load_img, img_to_array
+
+from pyspark import SparkContext
+from pyspark.sql import DataFrame, Row
+from pyspark.sql.functions import udf
+
+from sparkdl.graph.builder import IsolatedSession, GraphFunction
+import sparkdl.graph.utils as tfx
+
+from ..tests import SparkDLTestCase
+from ..transformers.image_utils import _getSampleJPEGDir, getSampleImagePathsDF
+
+
+class GraphBuilderTest(SparkDLTestCase):
```

Comment:
there is no longer a class called GraphBuilder, so rename to... `GraphFunctionWithIsolatedSessionTest` ?

several of the tests are testing GraphFunction and nearly all are testing IsolatedSession. It'd be ideal to separate them out, e.g. testing GraphFunction functions should use tf.Session. But the test coverage looks good enough so I'd add a TODO comment for this and call it a day for this PR (it's been modified enough...)